### PR TITLE
RDM-8598 - Integration tests for external search API

### PR DESCRIPTION
**Pull request by danlysiak**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-8598

### Change description ###
Implement Integration tests for external search API

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -256,6 +256,7 @@ dependencies {
     testCompile group: 'io.rest-assured', name: 'rest-assured', version: '3.1.0'
     testCompile group: 'com.xebialabs.restito', name: 'restito', version: '0.9.3'
     testCompile 'io.github.openfeign:feign-jackson:9.7.0'
+    testCompile 'pl.allegro.tech:embedded-elasticsearch:2.7.0'
```
Comment (smathangi) [inline build.gradle:259]: Not sure how good that embedded one. I prefer test containers in general https://www.testcontainers.org/modules/elasticsearch/

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+        void shouldReturnAllCasesForAllSpecifiedCaseTypes() throws Exception {
+            String searchRequest = ElasticsearchTestRequest.builder()
+                .query(matchAllQuery())
+                .build().toJsonString();
+
+            MvcResult result = mockMvc.perform(post(POST_SEARCH_CASES)
+                .contentType(MediaType.APPLICATION_JSON)
+                .param(CASE_TYPE_ID_PARAM, caseTypesParam(CASE_TYPE_A, CASE_TYPE_B))
+                .content(searchRequest))
+                .andExpect(status().is(200))
```
Comment (noronhaa) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java:76]: Is this test working as expected? - when trying this locally I get a 500 response with _Problem getting case type version for 'casetypeA,casetypeB'._

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+            );
+        }
+
+        public void assertExampleCaseMetadata(CaseDetails caseDetails) {
+            assertAll(
+                () -> assertThat(caseDetails.getJurisdiction(), is("AUTOTEST1")),
+                () -> assertThat(caseDetails.getCaseTypeId(), is(CASE_TYPE_A)),
+                () -> assertThat(caseDetails.getCreatedDate().toString(), is("2020-05-07T15:53:40.974")),
+                () -> assertThat(caseDetails.getLastModified().toString(), is("2020-06-09T13:17:06.542")),
+                // () -> assertThat(caseDetails.getLastStateModifiedDate().toString(), is("TBC")), // TODO: After RDM-8552 available
```
Comment (noronhaa) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java:232]: This should be available now so we should uncomment this line

Comment (danlysiak) [reply src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseSearchEndpointESIT.java]: Merged in latest and uncommented.

**Pull request merged**